OpenSim/Common/TimeSeriesTable.h

```diff
@@ -247,6 +246,46 @@ class TimeSeriesTable_ : public DataTable_<double, ETY> {
         *this = std::move(*table);
     }
 
+    /** Get index of row whose time is nearest/closest to the given value.
+
+    \param time Value to search for.
+    \param restrictToTimeRange  When true -- Exception is thrown if the given
+                                value is out-of-range of the time column. A value
+                                within SimTK::SignifcantReal of a time column
+                                bound is considered to be equal to the bound.
+                                When false -- If the given value is less than or
+                                equal to the first value in the time column, the
+                                index returned is of the first row. If the given
+                                value is greater than or equal to the last value
+                                in the time column, the index of the last row is
+                                returned. Defaults to 'true'.
+
+    \throws TimeOutOfRange If the given value is out-of-range of time column.
+    \throws EmptyTable If the table is empty.                                 */
+    size_t getNearestRowIndexForTime(const double time,
+                                     const bool restrictToTimeRange = true) const {
+        using DT = DataTable_<double, ETY>;
+        const auto& timeCol = DT::getIndependentColumn();
+        OPENSIM_THROW_IF(timeCol.size() == 0,
+            EmptyTable);
+        const SimTK::Real eps = SimTK::SignificantReal;
+        OPENSIM_THROW_IF(restrictToTimeRange &&
+            ((time < timeCol.front() - eps) ||
+            (time > timeCol.back() + eps)),
+            TimeOutOfRange,
+            time, timeCol.front(), timeCol.back());
+
+        auto iter = std::lower_bound(timeCol.begin(), timeCol.end(), time);
+        if (iter == timeCol.end())
+            return timeCol.size() - 1;
+        if (iter == timeCol.begin())
+            return 0;
```

Comment:
Not sure, but might generate a warning about implicit type conversion; perhaps `return 0u;` or `static_cast` to `size_t`?

Reply:
Ok. I see `return 0;` and automatically think "return with flag indicating that execution was successful" but yes I guess `0` will be interpreted as `0u` with no type conversion.